docs/userguide/constraints.md

```diff
@@ -222,10 +233,123 @@ DiscreteLinkedParametersConstraint(
+requires:
+1) A list ``parameters`` with the names of the parameters upon which others depend.
+2) A list of ``Condition``'s which specify the values of the corresponding entries in 
+   ``parameters`` which "activate" the dependent parameters.
+3) A list of lists, each containing the ``affected_parameters``, which become relevant 
+   only if the corresponding entry in ``parameters`` is active as specified by the entry in ``conditions``.
+
+Internally, BayBE drops entries from the ``SearchSpace`` where affected parameters are 
+irrelevant. Since in our example "off" is still a valid value for the switch, the 
+``SearchSpace`` will still retain an entry for that combination, showing arbitrary 
+values for the ``affected_parameters`` (which can be ignored).
+
+(DDC)=
+```{note}
+BayBE requires that all dependencies are declared in a single 
+``DiscreteDependenciesConstraint``, creating a ``SearchSpace`` from multiple 
+``DiscreteDependenciesConstraint``'s will throw a validation error.
+```
+
+In the example below we mimic a situation where there are two switches and each switch 
+activates two other parameters that are only relevant if the first switch is "on" / the 
+second switch is on "right" respectively.
+```python
+from baybe.constraints import DiscreteDependenciesConstraint, SubSelectionCondition
+
+DiscreteDependenciesConstraint(
+    parameters=["Switch1", "Switch2"],  # the two parameters upon which others depend
+    conditions=[
+        SubSelectionCondition(selection=["on"]),     # values of Switch1 that activate the affected parameters
+        SubSelectionCondition(selection=["right"]),  # values of Switch2 that activate the affected parameters
+    ],
+    affected_parameters=[
+      ["Solvent", "Fraction"],  # parameters affected by "Switch1"
+      ["Frame1", "Frame2"]      # parameters affected by "Switch2"
+    ],
+)
+```
+
+An end to end example can be found [here](../../examples/Constraints_Discrete/dependency_constraints).
 
 ### ``DiscretePermutationInvarianceConstraint``
-Content coming soon...
+Permutation invariance is a property where combinations of values of multiple
+parameters do not depend on their order due to some symmetry in the experiment. 
+Suppose that we are creating a mixture which has up to three solvents, i.e. parameters 
+"Solvent 1", "Solvent 2", "Solvent 3".
+In this situation, all combinations from the following table would be equivalent, 
+hence the ``SearchSpace`` should effectively only contain one of them.
+
+|   | Solvent 1    | Solvent 2    | Solvent 3    |
+|---|--------------|--------------|--------------|
+| 1 | Substance 43 | Substance 3  | Substance 12 |
+| 2 | Substance 43 | Substance 12 | Substance 3  |
+| 3 | Substance 3  | Substance 12 | Substance 43 |
+| 4 | Substance 3  | Substance 43 | Substance 12 |
+| 5 | Substance 12 | Substance 43 | Substance 3  |
+| 6 | Substance 12 | Substance 3  | Substance 43 |
+
+```{note}
```

Comment:
hmm, I explicitly did not make it specific because it would create a list that needs maintenance. Do we want that?

Reply:
Probably not, but maybe we should add a link to the most suitable page in the documentation?

Reply:
I can link the surrogate model page, but that doesnt really change anything regarding your original issue

maybe if the surrogate models all have text in the autodoc that explain it then it would work, but then for consistency we would prob need to add a lot of remakrs regarding different constraints into all surrogate models

Reply:
My point is that if not all surrogate models can include such variances, then we need to specify which can include them and which cannot at some place. This information is (as far as I am aware) currently not present anywhere.
The important question is then: Where should be we put this information? My feeling is that this information should rather be part of the corresponding surrogate page (or at least the surrogate user guide maybe?), hence my suggestion.
So: Do you agree that we should include this information more explicitly, and if so, where should we put it?

Reply:
I think this is going slightly off track because: what Martin explained in the text is that not all surrogates can handle these invariances. However, the wording probably needs to be changed a because I guess this might sound a bit like certain of our models **do automatically** handle these invariances. And I guess this is where Alex's request stems from. Yet, the truth is that e.g. GPs can handle them, but only if you build a custom kernel for it. So, out of the box, **NONE** of our surrogates handles them. 

So how about we change the order of statements:
1. First mention that these invariances should be ideally also handled during modelling
2. Out of the box, none of the shipped models do so
3. However, when designed properly, some of the models (such as GPs) can handle them. And here it's a bit difficult to give a precise list, because I also simply don't know how this could be encoded into other models. Perhaps there is also an easy way for other classes, or even a simple model-agnostic approach, but I guess that's then something the modeller should know!?

Reply:
Can confirm that I read this as "some surrogates can handle this while others can't".
Regarding your three points, I agree with the first two. I'd also be happy with a comment like "Although some of the models can handle invariances when modeled properly, this is not a naive feature of BayBE and requires the expertise of the user." and a statement to open an issue/contact us if this aspect is of importance for the user?

Reply:
yep I basically already wrote it from a perspective where our GP is updated. Ideally at soem point the GP discoverst atuomatically that permutaiton constraint is present (this is what I made the currently unused `eval_during_*` class properties for) and takes care of the kernels.

But to consolidate this issue here I changed the text to reflect the new situation, vote up if OK

Reply:
Only had a very minor comment regarding the formulation, wrote it at the corresponding line. Other then that, I am satisfied with the current version and grant thou my blessing for resolving.

Reply:
will resolve and discuss further in new comment